lib/acme_server/account.ex

```diff
@@ -1,12 +1,19 @@
 defmodule AcmeServer.Account do
+  @type site :: String.t()
+  @type dns :: %{String.t() => String.t()}
+  @type config :: %{site: site, site_uri: URI.t(), dns: dns}
+
+  @spec create(config, String.t()) :: map()
```

Comment:
`map()` is overly generic here. I think we need to introduce a type named `t` which describes the account map.